data_structures/src/superblock.rs

```diff
@@ -182,6 +183,21 @@ impl SuperBlockState {
             }
         }
     }
+
+    /// Returns an option with the last superblock hash and index if none of these fields is None.
+    /// else, returns None
+    pub fn get_beacon(&self) -> Option<CheckpointBeacon> {
+        match (
+            self.current_superblock_index.is_some(),
+            self.current_superblock_hash.is_some(),
+        ) {
+            (true, true) => Some(CheckpointBeacon {
+                checkpoint: self.current_superblock_index?,
+                hash_prev_block: self.current_superblock_hash?,
+            }),
+            _ => None,
+        }
```

Comment:
Actually, this is all you need. The match is redundant
```suggestion
            Some(CheckpointBeacon {
                checkpoint: self.current_superblock_index?,
                hash_prev_block: self.current_superblock_hash?,
            })
```